### PR TITLE
13 setup parser tests

### LINKED ISSUE #13: setup parser tests

### PR DESCRIPTION
**Pull request by vincent-lafouasse**
setup parser test and ensure proper linking

forward declaration of enums in `t_command.h` were also refactored to provide storage size quantifier (`enum my_enum : int;`)

### CODE REVIEW

**Comment by vincent-lafouasse** [comment]
> forward declaration of enums in t_command.h were also refactored to provide storage size quantifier (enum my_enum : int;)

this is no longer relevant

File: src/command/t_command.h
Comment (bitquence) [inline src/command/t_command.h:1]: should delete file

File: test/src/parser_test/CMakeLists.txt [hunk 1]
```diff
@@ -1,7 +1,43 @@
...
+)
+
+set(error_module
+	error/t_error.h
+	error/error_repr.c
+)
+set(log_module
+	log/log.c
+	log/log.h
+)
```
Comment (bitquence) [inline test/src/parser_test/CMakeLists.txt:28]: make sure to add json logging

Comment (bitquence) [reply test/src/parser_test/CMakeLists.txt:28]: lgtm otherwise

**Pull request merged**